activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -65,17 +65,20 @@ def eagerly_generate_alias_attribute_methods(_new_name, _old_name) # :nodoc:
       end
 
       def generate_alias_attributes # :nodoc:
-        return if @alias_attributes_mass_generated
+        klass = self
+        return if klass == ActiveRecord::Base
+        return klass.superclass.generate_alias_attributes if @alias_attributes_mass_generated
```

Comment:
It looks like we want to generate parent's alias attribute regardless of whether `@alias_attributes_mass_generated` is true or not. Should we just always call it at the first place like:
```ruby
klass.superclass.generate_alias_attributes
return if @alias_attributes_mass_generated
```
so we won't need to call it in the `synchronize` block?

Reply:
Good idea - updated in 217161c